datamapper/changeset.py

```diff
@@ -20,38 +21,81 @@
 from datamapper.model import Association, BelongsTo, Model
 
 FieldValidator = Callable[[Any], Union[bool, str]]
-Data = Union[dict, Model]
+Data = Union[Tuple[dict, dict], Model]
 
-T = TypeVar("T", dict, Model)
+T = TypeVar("T", Tuple[dict, dict], Model)
 
 
 class ChangesetDataWrapper(Generic[T]):
+    @staticmethod
+    def instance(data: T) -> ChangesetDataWrapper:
+        if isinstance(data, Model):
+            return ModelChangesetDataWrapper(data)
+        elif isinstance(data, tuple):
+            return DictChangesetDataWrapper(data)
+        else:
+            raise AttributeError("Changeset data must be a Model or (dict, dict).")
+
     def __init__(self, data: T):
-        self.data: T = data
+        self.data: T = data  # pragma: no cover
```

Comment:
You should probably just remove `__init__` here. It never gets called, and actually the `DictChangesetWrapper` doesn't take `data: T` , it takes `data: Tuple[T, dict]`